realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -794,4 +808,13 @@ public String toString() {
     public ProxyState realmGet$proxyState() {
         return proxyState;
     }
+
+    // Checks if the given field is primary key field. Throws if it is a PK field.
+    private void checkIsPrimaryKey(String fieldName) {
+        RealmObjectSchema objectSchema = proxyState.getRealm$realm().getSchema().getSchemaForClass(getType());
+        if (objectSchema.hasPrimaryKey() && objectSchema.getPrimaryKey().equals(fieldName)) {
+            throw new RealmException(String.format("Primary key field '%s' cannot be changed after object created."
```

Comment:
Typed Realm needs to throw as well. So throw IAE here and `RealmExcpetion` there? Will that look better?